3684/1/yaml_reader.rb

```diff
@@ -0,0 +1,52 @@
+require 'yaml'
+require 'gems'
+
+class YamlReader
+  attr_reader :file
+
+  def initialize(filepath)
+    @filepath = filepath
+  end
+
+  def parse
+    adress_of_source_code(read_yaml)
+  end
+
+  private
+
+  def read_yaml
+    file = YAML.load_file("#{@filepath}.yaml")
+    file['gems'].to_s.split(' ').each do |item|
+      item.delete!('-')
+    end
+  rescue IncorrectFileError
```

Comment:
you should catch `Errno` error class.
as I remember this error rises when file can't be open